CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 cmake_minimum_required(VERSION 3.13)
 project(stim)
 include_directories(src)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Happy to reverse this (and the use of `if consexpr` that requires it). Is the plan to update to cpp 20 as in #445?

Reply:
I am in general terrified of changing the standard. I am I think fine with the C++17 change, much more fine than the C++20 change because the manylinux standard(s) for python packages have some compatibility with C++17.

The C++20 PR is kind of just there to make sure I don't accidentally break forward compatibility, which has been an issue for some users in the past (e.g. I was using the `u8` prefix on strings wrong).